modules/registry/registry-core/src/main/java/org/apache/airavata/registry/core/repositories/expcatalog/ExperimentSummaryRepository.java

```diff
@@ -51,6 +56,17 @@
             throw new RegistryException("GatewayId is required");
         }
 
+        if (filters.get(DBConstants.Job.JOB_ID) != null) {
+            logger.debug("Filter Experiments by JobId");
+            queryParameters.put(DBConstants.Job.JOB_ID, filters.get(DBConstants.Job.JOB_ID));
+            String query_jobId = "SELECT P.experimentId FROM "
```

Comment:
Don't you need to filter by the gateway id as well?

Reply:
As discussed, the Airavata API is filtering the experiments that are accessible by the user using the sharing service.